chemclipse/plugins/org.eclipse.chemclipse.ux.extension.xxd.ui/src/org/eclipse/chemclipse/ux/extension/xxd/ui/swt/ExtendedTargetsUI.java

```diff
@@ -686,6 +686,7 @@ private void deleteTarget(ITarget target) {
 		if(object instanceof ITargetSupplier) {
 			ITargetSupplier targetSupplier = (ITargetSupplier)object;
 			targetSupplier.getTargets().remove(target);
+			((IChromatogram<?>)objectCacheChromatogram).setDirty(true);
```

Comment:
This might lead to a NPE, as `objectCacheChromatogram` could be null. A target supplier is not necessarily a chromatogram. It could be also a scan, a peak or another supplier.